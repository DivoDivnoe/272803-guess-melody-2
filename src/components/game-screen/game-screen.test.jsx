import React from 'react';
import renderer from 'react-test-renderer';
import GameScreen from './game-screen.jsx';

jest.mock(`../guess-artist-screen/guess-artist-screen.jsx`, () => jest.fn().mockReturnValue(null));
jest.mock(`../guess-genre-screen/guess-genre-screen.jsx`, () => jest.fn().mockReturnValue(null));
jest.mock(`../header/header.jsx`, () => jest.fn().mockReturnValue(null));

describe(`GameScreen component`, () => {
  it(`is rendered correctly`, () => {
    const question = {
      type: `artist`,
      song: {artist: ``, src: ``},
      answers: [{picture: `http://somesrc/`, artist: ``}]
    };
    const settings = {
      time: 30,
      mistakes: 2
    };
    const mistakes = 1;
    const gameTime = 0;
    const screenIndex = 0;

    const tree = renderer.create(
        <GameScreen
          question={question}
          mistakes={mistakes}
          gameTime={gameTime}
          screenIndex={screenIndex}
          settings={settings}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

