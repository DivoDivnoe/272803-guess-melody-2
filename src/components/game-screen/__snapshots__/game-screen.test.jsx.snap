// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameScreen component is rendered correctly 1`] = `
<section
  className="game game--artist"
/>
`;
